src/methods/gd_weighted_norm_damping.jl

```diff
@@ -0,0 +1,182 @@
+# Date: 11/14/2024
+# Author: Christian Varner
+# Purpose: Implement WNGrad
+
+"""
+    WeightedNormDampingGD{T} <: AbstractOptimizerData{T}
+
+A mutable struct that represents gradient descent using the weighted-norm 
+    damping  step size. It stores the specification for the method and records 
+    values during iteration.
+
+# Fields
+
+- `name::String`, name of the optimizer for recording purposes
+- `init_norm_damping_factor::T`, initial damping factor. This value's reciprocal 
+    will be the initial step size.
+- `threshold::T`, norm gradient tolerance condition. Induces stopping when norm 
+    at most `threshold`.
+- `max_iterations::Int64`, max number of iterates that are produced, not 
+    including the initial iterate.
+- `iter_hist::Vector{Vector{T}}`, store the iterate sequence as the algorithm 
+    progresses. The initial iterate is stored in the first position.
+- `grad_val_hist::Vector{T}`, stores the norm gradient values at each iterate. 
+    The norm of the gradient evaluated at the initial iterate is stored in the 
+    first position.
+- `stop_iteration::Int64`, the iteration number the algorithm stopped on. The 
+    iterate that induced stopping is saved at `iter_hist[stop_iteration + 1]`.
+
+# Constructors
+
+    WeightedNormDampingGD(::Type{T}; x0::Vector{T}, init_norm_damping_factor::T, 
+        threshold::T, max_iterations::Int64) where {T}
+
+Constructs an instance of type `WeightedNormDampingGD{T}`.
+
+## Arguments
+
+- `T::DataType`, type for data and computation
+
+## Keyword Arguments
+
+- `x0::Vector{T}`, initial point to start the optimization routine. Saved in
+    `iter_hist[1]`.
+- `init_norm_damping_factor::T`, initial damping factor, which will correspond
+    to the reciprocoal of the initial step size. 
+- `threshold::T`, norm gradient tolerance condition. Induces stopping when norm 
+    at most `threshold`.
+- `max_iterations::Int64`, max number of iterates that are produced, not 
+    including the initial iterate.
+"""
```

Comment:
Add a note about that the `init_norm_damping_factor` needs to be larger than the initial norm gradient @VChristian.